Cargo.lock

```diff
@@ -0,0 +1,28 @@
+[[package]]
+name = "libc"
+version = "0.2.66"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+
+[[package]]
+name = "pledge"
+version = "0.3.1"
+dependencies = [
+ "libc 0.2.66 (registry+https://github.com/rust-lang/crates.io-index)",
+]
+
+[[package]]
+name = "pledge-example-2015"
+version = "0.0.0"
+dependencies = [
+ "pledge 0.3.1",
+]
+
+[[package]]
+name = "pledge-example-2018"
+version = "0.0.0"
+dependencies = [
+ "pledge 0.3.1",
+]
+
+[metadata]
+"checksum libc 0.2.66 (registry+https://github.com/rust-lang/crates.io-index)" = "d515b1f41455adea1313a4a2ac8a8a477634fbae63cc6100e3aebb207ce61558"
```

Comment:
there might seem to be a contradiction between

1. Cargo.lock should only be checked in [for “end products” like binaries](https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html)
2. all targets in a crate and all members in a workspace share a Cargo.lock
3. workspaces/crates can mix members/targets around “end products”

but it turns out we can safely check in this Cargo.lock, because what Cargo actually does is ignore Cargo.lock when building a library target (at least as a dependency)

([FAQ](https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries) + [tweet](https://twitter.com/dazabani/status/1167058283191926784))